scripts/build_failure_data_collection.py

```diff
@@ -0,0 +1,294 @@
+
+    for task in queue_client.listTaskGroup(task_group_id)["tasks"]:
+        if task["status"]["state"] == "failed":
+            failed_tasks.add(task["status"]["taskId"])
+
+    # 6. find intersection between build tasks and failed tasks
+    failed_build_tasks = list(build_tasks & failed_tasks)
+
+    # 7. get the url to access the log, load it, and extract the ERROR lines
+    error_lines = []
+
+    for failed_build_task in failed_build_tasks:
+        artifact = queue_client.getArtifact(
+            taskId=failed_build_task, runId="0", name="public/logs/live.log"
+        )
+        url = artifact["url"]
+
+        response = requests.get(url)
+        error_lines.extend(
+            [line for line in response.text.split("\n") if "ERROR - " in line]
+        )
+
+    return error_lines
+
+
+def main():
+    # 0.
+    download_databases()
+
+    # 1.
+    bugs = get_backed_out_build_failure_bugs()
+    bug_ids = list(bugs.keys())
+
+    # 2.
+    bug_commits = map_bugs_to_commit(bug_ids)
+
+    # 3.
+    hg_client = Revision()
+    backed_out_revisions = find_bugs(hg_client, bug_ids, bug_commits)
+
+    # 4.
+    revisions_to_commits = defaultdict(list)
+
+    for commit in repository.get_commits():
+        revision_id = repository.get_revision_id(commit)
+
+        if revision_id in backed_out_revisions:
+            revisions_to_commits[revision_id].append(commit["node"])
+
+    # 5. and 6.
+
+    client_id = os.getenv("TC_CLIENT_ID")
+
+    index = taskcluster.Index(
+        {
+            "rootUrl": "https://firefox-ci-tc.services.mozilla.com",
+            "credentials": {"clientId": client_id},
+        }
+    )
+
+    queue = taskcluster.Queue(
+        {
+            "rootUrl": "https://firefox-ci-tc.services.mozilla.com",
+        }
+    )
+
+    with open("revisions.csv", mode="w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+
+        writer.writerow(
+            ["Revision ID", "Initial Commit", "Fix Commit", "Interdiff", "Error Lines"]
+        )
+
+        for revision_id, commits in revisions_to_commits.items():
+            if len(commits) < 2:
+                continue
+
+            for commit in commits:
+                error_lines = find_error_lines(index, queue, commit)
+
+                if error_lines:
+                    break
+
+            # if not error_lines:
+            #     continue
+
+            commit_diff = repository.get_diff(
+                repo_path="hg_dir", original_hash=commits[0], fix_hash=commits[1]
+            )
+            # if not commit_diff:
+            #     continue
+
+            commit_diff_encoded = commit_diff.decode("utf-8", errors="replace")
+
+            writer.writerow(
+                [revision_id, commits[0], commits[1], commit_diff_encoded, error_lines]
+            )
+
+
+if __name__ == "__main__":
+    main()
+
+# 0. Download databases
+# 1. Identify bugs in Bugzilla that have a backout due to build failures X
```

Comment:
I normally get my regression (backed out) data from treeherder database.  The main reason why is we can get all the tasks that failed, which could help determine if this is failing ALL builds, or a certain platform or build failure.  Not required, but the additional information is good to know.